src/nidcpower/metadata/config.py

```diff
@@ -27,5 +27,6 @@
     'repeated_capabilities': [
         {'python_name': 'channels', 'prefix': '', },
     ],
+    'use_session_lock': True,
```

Comment:
I don't think this is being used. Would be better not to have an extra piece of global metadata unless there's no better way.

Reply:
It is being used during add_all to know what the default should be. I.e. if it isn't already set for a function, the value from config is uses.

Reply:
But if it doesn't need to be configurable, then we should remove it and default to True.

Or are you concerned that processed metadata for nimodinst will say True everywhere which is misleading? Is this enough reason to add config metadata?

Alternatively, and assuming there are potential race conditions in nimodinst bindings because several functions in the library are called within a public method call,  we could fix nimodinst race conditions by adding our own recursive lock at the Python level (I don't favor this but we could at least document it as an issue).

Reply:
Saying `True` for nimodinst isn't just misleading, but will generate incorrect code. It will then try to add the lock decorator for each function in #857